src/clj/rems/db/api_key.clj

```diff
@@ -1,5 +1,14 @@
 (ns rems.db.api-key
-  (:require [rems.db.core :as db]))
+  (:require [rems.db.core :as db]
+            [rems.json :as json]))
 
 (defn valid? [key]
   (not (nil? (db/get-api-key {:apikey key}))))
+
+(defn add-api-key! [key unavailable-roles comment]
+  (db/add-api-key! {:apikey key
+                    :comment comment
+                    :unavailableroles (json/generate-string unavailable-roles)}))
```

Comment:
why a blacklist and not a whitelist?

Reply:
whitelists are typically better for security stuff like this. otherwise when we add a new role (e.g. domainowner) it will "leak" to all existing api keys when upgrading REMS

Reply:
Right, good point. I'll change it to a whitelist.